client/templates/layouts/core.js

```diff
@@ -12,45 +14,62 @@ class CoreLayoutBeesknees extends Component {
     structure: PropTypes.object
   }
 
+  getComponent(name) {
+    try {
+      if (name) {
+        return assertComponent(name);
+      }
+    } catch (e) { }
+    return null;
+  }
+
+  renderMain() {
+    const template = this.props.structure && this.props.structure.template;
+    let mainComponent = this.getComponent(template);
+    if (mainComponent) {
+      return React.createElement(mainComponent, {});
+    }
+    else if (Template[template]) {
+      return <Blaze template={template} />;
+    }
+    return null;
+  }
+
   render() {
     const { layoutHeader, layoutFooter, template } = this.props.structure || {};
     const pageClassName = classnames({
       "page": true,
       "show-settings": this.props.actionViewIsOpen
     });
 
+    const headerComponent = layoutHeader && this.getComponent(layoutHeader);
+    const footerComponent = layoutFooter && this.getComponent(layoutFooter);
+
     return (
       <div className={pageClassName} id="reactionAppContainer">
-        { Template[layoutHeader] &&
-          <Blaze template={layoutHeader} className="reaction-navigation-header" />
-        }
+
+        { headerComponent && React.createElement(headerComponent, {}) }
```

Comment:
just a nit-picky-style-thing, but I would have extra spaces with `{}` unless the linter told me otherwise.

Reply:
you probably meant "I would have *no* extra spaces" , no?
At least I've read that in another comment, if I remember correctly..